packages/flutter/lib/src/services/autofill.dart

```diff
@@ -0,0 +1,273 @@
+  /// on Android, and "tel" on web.
+  static const String telephoneNumber = 'telephoneNumber';
+
+  /// The client represents an input field that expects the first line of a street
+  /// address.
+  ///
+  /// Translates to [UITextContentType.streetAddressLine1](https://developer.apple.com/documentation/uikit/uitextcontenttype/1649663-streetaddressline1),
+  /// on iOS, [.AUTOFILL_HINT_POSTAL_ADDRESS](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_POSTAL_ADDRESS)
+  /// on Android, and "address-line1" on web.
+  static const String streetAddressLine1 = 'streetAddressLine1';
+
+  /// The client represents an input field that expects the second line of a street
+  /// address.
+  ///
+  /// Translates to [UITextContentType.streetAddressLine2](https://developer.apple.com/documentation/uikit/uitextcontenttype/1649658-streetaddressline2),
+  /// on iOS, [.AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_ADDRESS](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_ADDRESS)
+  /// on Android, and "address-line2" on web.
+  static const String streetAddressLine2 = 'streetAddressLine2';
+
+  /// The client represents a password field.
+  ///
+  /// Translates to [UITextContentType.password](https://developer.apple.com/documentation/uikit/uitextcontenttype/2865813-password),
+  /// on iOS, [.AUTOFILL_HINT_PASSWORD](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_PASSWORD)
+  /// on Android, and ""current-password" on web.
+  static const String password = 'password';
+
+  /// The client represents a username field or an account name.
+  ///
+  /// Translates to [UITextContentType.username](https://developer.apple.com/documentation/uikit/uitextcontenttype/2866088-username),
+  /// on iOS, [.AUTOFILL_HINT_USERNAME](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_USERNAME)
+  /// on Android, and "username" on web.
+  static const String username = 'username';
+}
+
+/// A collection of autofill related information that represents an [AutofillClient].
+///
+/// Typically used in [TextInputConfiguration.autofillConfiguration].
+@immutable
+class AutofillConfiguration {
+  /// Creates autofill related configuration information that can be sent to the
+  /// platform.
+  const AutofillConfiguration({
+    @required this.uniqueIdentifier,
+    @required this.autofillHints,
+    this.currentEditingValue,
+  }) : assert(uniqueIdentifier != null),
+       assert(autofillHints != null);
+
+  /// A string that uniquely identifies the current [AutofillClient].
+  ///
+  /// The identifier needs to be unique within the [AutofillScope] for the
+  /// [AutofillClient] to receive the correct autofill value.
+  ///
+  /// Must not be null. If left unspecified, a random string will be used.
```

Comment:
Is it true that a random string will be used? Above, it asserts that `uniqueIdentifier` is not null and doesn't default it to anything.